codegens/csharp-restsharp/npm/test-lint.js

```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env node
-require('shelljs/global');
+require('shelljs');
```

Comment:
After this change, I am not able to run `npm run test-lint` command inside any codegen. It states
`ReferenceError: exit is not defined
    at Object.<anonymous> (/Users/shreyshah/Desktop/open-source/code-generators/codegens/csharp-restsharp/npm/test-lint.js:59:34)`
. Thus, assign it to a variable and use shell.exit. This change needs to be done in all codegens lint tests.

Reply:
Forgot to check lint on codegens and there is no such script on main package.json also to check lint recursively. Isn't there should such script and moreover I guess CI should also check lint before passing tests. Fixed issue for now the test is passing on local as follows:
`
nikhil@Nikhil-PC:/media/nikhil/Data/Projects/postman-code-generators/codegens$ ./a.sh > a.txt

> @postman/codegen-csharp-restsharp@0.1.1 test-lint /media/nikhil/Data/Projects/postman-code-generators/codegens/csharp-restsharp
> node npm/test-lint.js


Linting files using eslint...

/media/nikhil/Data/Projects/postman-code-generators/codegens/csharp-restsharp/lib/lodash.js
  345:27  warning  Found non-literal argument to RegExp Constructor  security/detect-non-literal-regexp

/media/nikhil/Data/Projects/postman-code-generators/codegens/csharp-restsharp/npm/test-newman.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

/media/nikhil/Data/Projects/postman-code-generators/codegens/csharp-restsharp/npm/test-unit.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

✖ 3 problems (0 errors, 3 warnings)

eslint ok!

> @postman/codegen-curl@0.1.6 test-lint /media/nikhil/Data/Projects/postman-code-generators/codegens/curl
> node npm/test-lint.js


Linting files using eslint...

/media/nikhil/Data/Projects/postman-code-generators/codegens/curl/lib/lodash.js
  346:27  warning  Found non-literal argument to RegExp Constructor  security/detect-non-literal-regexp

/media/nikhil/Data/Projects/postman-code-generators/codegens/curl/npm/test-newman.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

/media/nikhil/Data/Projects/postman-code-generators/codegens/curl/npm/test-unit.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

✖ 3 problems (0 errors, 3 warnings)

eslint ok!

> @postman/codegen-golang@0.1.6 test-lint /media/nikhil/Data/Projects/postman-code-generators/codegens/golang
> node npm/test-lint.js


Linting files using eslint...

/media/nikhil/Data/Projects/postman-code-generators/codegens/golang/lib/lodash.js
  346:27  warning  Found non-literal argument to RegExp Constructor  security/detect-non-literal-regexp

/media/nikhil/Data/Projects/postman-code-generators/codegens/golang/npm/test-newman.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

/media/nikhil/Data/Projects/postman-code-generators/codegens/golang/npm/test-unit.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

✖ 3 problems (0 errors, 3 warnings)

eslint ok!

> @postman/codegen-http@0.1.1 test-lint /media/nikhil/Data/Projects/postman-code-generators/codegens/http
> node npm/test-lint.js


Linting files using eslint...

/media/nikhil/Data/Projects/postman-code-generators/codegens/http/lib/lodash.js
  346:27  warning  Found non-literal argument to RegExp Constructor  security/detect-non-literal-regexp

/media/nikhil/Data/Projects/postman-code-generators/codegens/http/npm/test-unit.js
  34:30  warning  Expected to return a value at the end of function  consistent-return

✖ 2 problems (0 errors, 2 warnings)

eslint ok!

> @postman/codegen-java-okhttp@0.1.2 test-lint /media/nikhil/Data/Projects/postman-code-generators/codegens/java-okhttp
> node npm/test-lint.js


Linting files using eslint...

/media/nikhil/Data/Projects/postman-code-generators/codegens/java-okhttp/lib/lodash.js
  345:27  warning  Found non-literal argument to RegExp Constructor  security/detect-non-literal-regexp

/media/nikhil/Data/Projects/postman-code-generators/codegens/java-okhttp/npm/test-newman.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

/media/nikhil/Data/Projects/postman-code-generators/codegens/java-okhttp/npm/test-unit.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

✖ 3 problems (0 errors, 3 warnings)

eslint ok!

> @postman/codegen-java-unirest@0.1.1 test-lint /media/nikhil/Data/Projects/postman-code-generators/codegens/java-unirest
> node npm/test-lint.js


Linting files using eslint...

/media/nikhil/Data/Projects/postman-code-generators/codegens/java-unirest/lib/lodash.js
  346:27  warning  Found non-literal argument to RegExp Constructor  security/detect-non-literal-regexp

/media/nikhil/Data/Projects/postman-code-generators/codegens/java-unirest/npm/test-newman.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

/media/nikhil/Data/Projects/postman-code-generators/codegens/java-unirest/npm/test-unit.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

✖ 3 problems (0 errors, 3 warnings)

eslint ok!

> @postman/codegen-js-fetch@0.1.1 test-lint /media/nikhil/Data/Projects/postman-code-generators/codegens/js-fetch
> node npm/test-lint.js


Linting files using eslint...

/media/nikhil/Data/Projects/postman-code-generators/codegens/js-fetch/lib/lodash.js
  346:27  warning  Found non-literal argument to RegExp Constructor  security/detect-non-literal-regexp

/media/nikhil/Data/Projects/postman-code-generators/codegens/js-fetch/npm/test-newman.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

/media/nikhil/Data/Projects/postman-code-generators/codegens/js-fetch/npm/test-unit.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

✖ 3 problems (0 errors, 3 warnings)

eslint ok!

> @postman/codegen-js-jquery@0.1.4 test-lint /media/nikhil/Data/Projects/postman-code-generators/codegens/js-jquery
> node npm/test-lint.js


Linting files using eslint...

/media/nikhil/Data/Projects/postman-code-generators/codegens/js-jquery/lib/lodash.js
  346:27  warning  Found non-literal argument to RegExp Constructor  security/detect-non-literal-regexp

/media/nikhil/Data/Projects/postman-code-generators/codegens/js-jquery/npm/test-unit.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

✖ 2 problems (0 errors, 2 warnings)

eslint ok!

> @postman/codegen-js-xhr@0.1.1 test-lint /media/nikhil/Data/Projects/postman-code-generators/codegens/js-xhr
> node npm/test-lint.js


Linting files using eslint...

/media/nikhil/Data/Projects/postman-code-generators/codegens/js-xhr/lib/lodash.js
  346:27  warning  Found non-literal argument to RegExp Constructor  security/detect-non-literal-regexp

/media/nikhil/Data/Projects/postman-code-generators/codegens/js-xhr/npm/test-newman.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

/media/nikhil/Data/Projects/postman-code-generators/codegens/js-xhr/npm/test-unit.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

✖ 3 problems (0 errors, 3 warnings)

eslint ok!

> @postman/codegen-libcurl@0.1.0 test-lint /media/nikhil/Data/Projects/postman-code-generators/codegens/libcurl
> node npm/test-lint.js


Linting files using eslint...

/media/nikhil/Data/Projects/postman-code-generators/codegens/libcurl/lib/lodash.js
  354:27  warning  Found non-literal argument to RegExp Constructor  security/detect-non-literal-regexp

/media/nikhil/Data/Projects/postman-code-generators/codegens/libcurl/npm/test-newman.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

/media/nikhil/Data/Projects/postman-code-generators/codegens/libcurl/npm/test-unit.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

✖ 3 problems (0 errors, 3 warnings)

eslint ok!

> @postman/codegen-nodejs-native@0.1.3 test-lint /media/nikhil/Data/Projects/postman-code-generators/codegens/nodejs-native
> node npm/test-lint.js


Linting files using eslint...

/media/nikhil/Data/Projects/postman-code-generators/codegens/nodejs-native/lib/lodash.js
  346:27  warning  Found non-literal argument to RegExp Constructor  security/detect-non-literal-regexp

/media/nikhil/Data/Projects/postman-code-generators/codegens/nodejs-native/npm/test-newman.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

/media/nikhil/Data/Projects/postman-code-generators/codegens/nodejs-native/npm/test-unit.js
  36:30  warning  Expected to return a value at the end of function  consistent-return

✖ 3 problems (0 errors, 3 warnings)

eslint ok!

> @postman/codegen-nodejs-request@0.1.2 test-lint /media/nikhil/Data/Projects/postman-code-generators/codegens/nodejs-request
> node npm/test-lint.js


Linting files using eslint...

/media/nikhil/Data/Projects/postman-code-generators/codegens/nodejs-request/lib/lodash.js
  346:27  warning  Found non-literal argument to RegExp Constructor  security/detect-non-literal-regexp

/media/nikhil/Data/Projects/postman-code-generators/codegens/nodejs-request/npm/test-newman.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

/media/nikhil/Data/Projects/postman-code-generators/codegens/nodejs-request/npm/test-unit.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

✖ 3 problems (0 errors, 3 warnings)

eslint ok!

> @postman/codegen-nodejs-unirest@0.1.1 test-lint /media/nikhil/Data/Projects/postman-code-generators/codegens/nodejs-unirest
> node npm/test-lint.js


Linting files using eslint...

/media/nikhil/Data/Projects/postman-code-generators/codegens/nodejs-unirest/lib/lodash.js
  346:27  warning  Found non-literal argument to RegExp Constructor  security/detect-non-literal-regexp

/media/nikhil/Data/Projects/postman-code-generators/codegens/nodejs-unirest/npm/test-newman.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

/media/nikhil/Data/Projects/postman-code-generators/codegens/nodejs-unirest/npm/test-unit.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

✖ 3 problems (0 errors, 3 warnings)

eslint ok!

> @postman/codegen-objective-c@0.0.1 test-lint /media/nikhil/Data/Projects/postman-code-generators/codegens/objective-c
> node npm/test-lint.js


Linting files using eslint...

/media/nikhil/Data/Projects/postman-code-generators/codegens/objective-c/lib/lodash.js
  345:27  warning  Found non-literal argument to RegExp Constructor  security/detect-non-literal-regexp

/media/nikhil/Data/Projects/postman-code-generators/codegens/objective-c/npm/test-newman.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

/media/nikhil/Data/Projects/postman-code-generators/codegens/objective-c/npm/test-unit.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

✖ 3 problems (0 errors, 3 warnings)

eslint ok!

> @postman/codegen-ocaml-cohttp@0.0.1 test-lint /media/nikhil/Data/Projects/postman-code-generators/codegens/ocaml-cohttp
> node npm/test-lint.js


Linting files using eslint...

/media/nikhil/Data/Projects/postman-code-generators/codegens/ocaml-cohttp/lib/lodash.js
  345:27  warning  Found non-literal argument to RegExp Constructor  security/detect-non-literal-regexp

/media/nikhil/Data/Projects/postman-code-generators/codegens/ocaml-cohttp/npm/test-unit.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

✖ 2 problems (0 errors, 2 warnings)

eslint ok!

> @postman/codegen-php-curl@0.1.4 test-lint /media/nikhil/Data/Projects/postman-code-generators/codegens/php-curl
> node npm/test-lint.js


Linting files using eslint...

/media/nikhil/Data/Projects/postman-code-generators/codegens/php-curl/lib/lodash.js
  346:27  warning  Found non-literal argument to RegExp Constructor  security/detect-non-literal-regexp

/media/nikhil/Data/Projects/postman-code-generators/codegens/php-curl/lib/util/parseBody.js
  55:44  warning  Expected to return a value at the end of function  consistent-return

/media/nikhil/Data/Projects/postman-code-generators/codegens/php-curl/npm/test-newman.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

/media/nikhil/Data/Projects/postman-code-generators/codegens/php-curl/npm/test-unit.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

✖ 4 problems (0 errors, 4 warnings)

eslint ok!

> @postman/codegen-php-httprequest2@0.0.1 test-lint /media/nikhil/Data/Projects/postman-code-generators/codegens/php-httprequest2
> node npm/test-lint.js


Linting files using eslint...

/media/nikhil/Data/Projects/postman-code-generators/codegens/php-httprequest2/lib/lodash.js
  345:27  warning  Found non-literal argument to RegExp Constructor  security/detect-non-literal-regexp

/media/nikhil/Data/Projects/postman-code-generators/codegens/php-httprequest2/npm/test-newman.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

/media/nikhil/Data/Projects/postman-code-generators/codegens/php-httprequest2/npm/test-unit.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

✖ 3 problems (0 errors, 3 warnings)

eslint ok!

> @postman/codegen-php-pecl-http@0.1.1 test-lint /media/nikhil/Data/Projects/postman-code-generators/codegens/php-pecl-http
> node npm/test-lint.js


Linting files using eslint...

/media/nikhil/Data/Projects/postman-code-generators/codegens/php-pecl-http/lib/lodash.js
  346:27  warning  Found non-literal argument to RegExp Constructor  security/detect-non-literal-regexp

/media/nikhil/Data/Projects/postman-code-generators/codegens/php-pecl-http/npm/test-unit.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

✖ 2 problems (0 errors, 2 warnings)

eslint ok!

> @postman/codegen-powershell-restmethod@0.0.1 test-lint /media/nikhil/Data/Projects/postman-code-generators/codegens/powershell-restmethod
> node npm/test-lint.js


Linting files using eslint...

/media/nikhil/Data/Projects/postman-code-generators/codegens/powershell-restmethod/lib/lodash.js
  346:27  warning  Found non-literal argument to RegExp Constructor  security/detect-non-literal-regexp

/media/nikhil/Data/Projects/postman-code-generators/codegens/powershell-restmethod/test/unit/convert.test.js
  118:10  warning  Unexpected skipped mocha test  mocha/no-skipped-tests

/media/nikhil/Data/Projects/postman-code-generators/codegens/powershell-restmethod/npm/test-unit.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

✖ 3 problems (0 errors, 3 warnings)
  0 errors, 1 warning potentially fixable with the `--fix` option.

eslint ok!

> @postman/codegen-python-http.client@0.1.1 test-lint /media/nikhil/Data/Projects/postman-code-generators/codegens/python-http.client
> node npm/test-lint.js


Linting files using eslint...

/media/nikhil/Data/Projects/postman-code-generators/codegens/python-http.client/lib/lodash.js
  345:27  warning  Found non-literal argument to RegExp Constructor  security/detect-non-literal-regexp

/media/nikhil/Data/Projects/postman-code-generators/codegens/python-http.client/npm/test-newman.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

/media/nikhil/Data/Projects/postman-code-generators/codegens/python-http.client/npm/test-unit.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

✖ 3 problems (0 errors, 3 warnings)

eslint ok!

> @postman/codegen-python-requests@0.1.1 test-lint /media/nikhil/Data/Projects/postman-code-generators/codegens/python-requests
> node npm/test-lint.js


Linting files using eslint...

/media/nikhil/Data/Projects/postman-code-generators/codegens/python-requests/lib/lodash.js
  346:27  warning  Found non-literal argument to RegExp Constructor  security/detect-non-literal-regexp

/media/nikhil/Data/Projects/postman-code-generators/codegens/python-requests/npm/test-newman.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

/media/nikhil/Data/Projects/postman-code-generators/codegens/python-requests/npm/test-unit.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

✖ 3 problems (0 errors, 3 warnings)

eslint ok!

> @postman/codegen-ruby@0.1.1 test-lint /media/nikhil/Data/Projects/postman-code-generators/codegens/ruby
> node npm/test-lint.js


Linting files using eslint...

/media/nikhil/Data/Projects/postman-code-generators/codegens/ruby/lib/lodash.js
  346:27  warning  Found non-literal argument to RegExp Constructor  security/detect-non-literal-regexp

/media/nikhil/Data/Projects/postman-code-generators/codegens/ruby/npm/test-newman.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

/media/nikhil/Data/Projects/postman-code-generators/codegens/ruby/npm/test-unit.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

✖ 3 problems (0 errors, 3 warnings)

eslint ok!

> @postman/codegen-shell-httpie@0.1.0 test-lint /media/nikhil/Data/Projects/postman-code-generators/codegens/shell-httpie
> ./scripts/test-lint.sh

[93mLinting and style-checking...[0m
[0m[2meslint v4.18.2[0m


/media/nikhil/Data/Projects/postman-code-generators/codegens/shell-httpie/lib/lodash.js
  346:27  warning  Found non-literal argument to RegExp Constructor  security/detect-non-literal-regexp

/media/nikhil/Data/Projects/postman-code-generators/codegens/shell-httpie/lib/shell-httpie.js
  50:12  warning  Expected to return a value at the end of method 'convert'  consistent-return

✖ 2 problems (0 errors, 2 warnings)

[92mNo lint errors found.
[0m
> @postman/codegen-shell-wget@0.1.2 test-lint /media/nikhil/Data/Projects/postman-code-generators/codegens/shell-wget
> node npm/test-lint.js


Linting files using eslint...

/media/nikhil/Data/Projects/postman-code-generators/codegens/shell-wget/lib/lodash.js
  345:27  warning  Found non-literal argument to RegExp Constructor  security/detect-non-literal-regexp

/media/nikhil/Data/Projects/postman-code-generators/codegens/shell-wget/npm/test-newman.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

/media/nikhil/Data/Projects/postman-code-generators/codegens/shell-wget/npm/test-unit.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

✖ 3 problems (0 errors, 3 warnings)

eslint ok!

> @postman/codegen-swift@0.0.1 test-lint /media/nikhil/Data/Projects/postman-code-generators/codegens/swift
> node npm/test-lint.js


Linting files using eslint...

/media/nikhil/Data/Projects/postman-code-generators/codegens/swift/lib/lodash.js
  345:27  warning  Found non-literal argument to RegExp Constructor  security/detect-non-literal-regexp

/media/nikhil/Data/Projects/postman-code-generators/codegens/swift/npm/test-newman.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

/media/nikhil/Data/Projects/postman-code-generators/codegens/swift/npm/test-unit.js
  38:30  warning  Expected to return a value at the end of function  consistent-return

✖ 3 problems (0 errors, 3 warnings)

eslint ok!
`

Reply:
Raised Issue #203 for this issue.